[CWS] Keep track of container runtime and cgroup id (#26425)
DataDog/datadog-agent: docs/cloud-workload-security/agent_expressions.md

<<<<<<< HEAD
[1]: /security/threats/workload_security_rules/custom_rules
[2]: /security/threats/linux_expressions
[3]: /security/threats/windows_expressions
=======
{{< code-block lang="javascript" >}}
open.file.path == "/proc/${process.pid}/maps"

{{< /code-block >}}

List of the available variables:

| SECL Variable         |  Definition                           | Agent Version |
|-----------------------|---------------------------------------|---------------|
| `process.pid`         | Process PID                           | 7.33          |

## CIDR and IP range
CIDR and IP matching is possible in SECL. One can use operators such as `in`, `not in`, or `allin` combined with CIDR or IP notations.

Such rules can be written as follows:

{{< code-block lang="javascript" >}}
dns.question.name == "example.com" && network.destination.ip in [192.168.1.25, 10.0.0.0/24]

{{< /code-block >}}

## Helpers
Helpers exist in SECL that enable users to write advanced rules without needing to rely on generic techniques such as regex.

### Command line arguments
The *args_flags* and *args_options* are helpers to ease the writing of CSM Threats rules based on command line arguments.

> *args_flags* is used to catch arguments that start with either one or two hyphen characters but do not accept any associated value.

Reviewer: ```suggestion
*args_flags* is used to catch arguments that start with either one or two hyphen characters but do not accept any associated values.
```

Examples:
* `version` is part of *args_flags* for the command `cat --version`
* `l` and `n` both are in *args_flags* for the command `netstat -ln`

> *args_options* is used to catch arguments that start with either one or two hyphen characters and accepts a value either specified as the same argument but separated by the ‘=’ character or specified as the next argument.

Reviewer: ```suggestion
*args_options* is used to catch arguments that start with either one or two hyphen characters. It accepts values that are specified as the same argument (but separated by the `=` character) or as the next argument.
```

Examples:
* `T=8` and `width=8` both are in *args_options* for the command `ls -T 8 --width=8`
* `exec.args_options in [ r"s=.*\\" ]` can be used to detect `sudoedit` was launched with `-s` argument and a command that ends with a `\`

### File rights

> The *file.rights* attribute can now be used in addition to *file.mode*. *file.mode* can hold values set by the kernel, while the *file.rights* only holds the values set by the user. These rights may be more familiar because they are in the `chmod` commands.

Reviewer: ```suggestion
You can use the *file.rights* attribute in addition to *file.mode*. *file.mode* can hold values set by the kernel, while *file.rights* only holds the values set by the user. These rights may be more familiar because they are in the `chmod` commands.
```

| [`utimes.file.mount_id`](#common-pathkey-mount_id-doc) | Mount ID of the file |
| [`utimes.file.name`](#common-fileevent-name-doc) | File's basename |
| [`utimes.file.name.length`](#common-string-length-doc) | Length of the corresponding string |
| [`utimes.file.package.name`](#common-fileevent-package-name-doc) | [Experimental] Name of the package that provided this file |
| [`utimes.file.package.source_version`](#common-fileevent-package-source_version-doc) | [Experimental] Full version of the source package of the package that provided this file |
| [`utimes.file.package.version`](#common-fileevent-package-version-doc) | [Experimental] Full version of the package that provided this file |
| [`utimes.file.path`](#common-fileevent-path-doc) | File's path |
| [`utimes.file.path.length`](#common-string-length-doc) | Length of the corresponding string |
| [`utimes.file.rights`](#common-filefields-rights-doc) | Rights of the file |
| [`utimes.file.uid`](#common-filefields-uid-doc) | UID of the file's owner |
| [`utimes.file.user`](#common-filefields-user-doc) | User of the file's owner |
| [`utimes.retval`](#common-syscallevent-retval-doc) | Return value of the syscall |

> ## Attributes documentation

Reviewer: ```suggestion
## Attributes
```

| `SIGPWR` | all |
| `SIGSYS` | all |

### `Unlink flags` {#unlink-flags}
Unlink flags are the supported flags for the unlink syscall.

| Name | Architectures |
| ---- |---------------|
| `AT_REMOVEDIR` | all |

### `Virtual Memory flags` {#virtual-memory-flags}
Virtual Memory flags define the protection of a virtual memory segment.

| Name | Architectures |
| ---- |---------------|
| `VM_NONE` | all |
| `VM_READ` | all |
| `VM_WRITE` | all |
| `VM_EXEC` | all |
| `VM_SHARED` | all |
| `VM_MAYREAD` | all |
| `VM_MAYWRITE` | all |
| `VM_MAYEXEC` | all |
| `VM_MAYSHARE` | all |
| `VM_GROWSDOWN` | all |
| `VM_UFFD_MISSING` | all |
| `VM_PFNMAP` | all |
| `VM_UFFD_WP` | all |
| `VM_LOCKED` | all |
| `VM_IO` | all |
| `VM_SEQ_READ` | all |
| `VM_RAND_READ` | all |
| `VM_DONTCOPY` | all |
| `VM_DONTEXPAND` | all |
| `VM_LOCKONFAULT` | all |
| `VM_ACCOUNT` | all |
| `VM_NORESERVE` | all |
| `VM_HUGETLB` | all |
| `VM_SYNC` | all |
| `VM_ARCH_1` | all |
| `VM_WIPEONFORK` | all |
| `VM_DONTDUMP` | all |
| `VM_SOFTDIRTY` | all |
| `VM_MIXEDMAP` | all |
| `VM_HUGEPAGE` | all |
| `VM_NOHUGEPAGE` | all |
| `VM_MERGEABLE` | all |

{{< partial name="whats-next/whats-next.html" >}}

> >>>>>>> 4f4578f5fd (Export container.flags to SECL)

Reviewer: more merge conflict errata
